---
title: "Chromatin Data Association"
author: "Ruben Schep"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  html_document:
    theme: journal #cerulean
    highlight: monochrome
    toc: true
    toc_float: true
    code_folding: show
  editor_options:
    chunk_output_type: console
---

# knitr document van Steensel lab

# Introduction
After having done the preprocessing of the mapping and the indels, we want to associate the indels with different chromatin marks.
1. We will use the ChIP seq data from ENCODE, get the ChIP signal over 1kb of each integration region.
2. We will get LAD calling for each integration. And distance to LAD border.
3. We want the 5kb bin signal for for each integration for LmnB1, 3xAP3 and ?? from Tom's DamID.
4. We want gene information (in or out of a gene, distance to TSS, transcriptional data of those genes)
5. We want chromatin accessibility, with DHS signal.
6. We want compartments
7. TAD structure could be interesting. TAD border?
8. We can do domain calling with HiddenDomains from the ChIP seq data and look at domain location and distance.



## Description of Data
We will start with the mapping data that was preprocessed and contains about 1900 genomic locations. Note, we should still add mapped to repeats, that might be interesting for the data.


# Data processing
## Path, Libraries, Parameters and Useful Functions
```{r setup}
knitr::opts_chunk$set(echo = TRUE)
StartTime <-Sys.time()

# 6-digit Date tag:
Date <- substr(gsub("-","",Sys.time()),1,8)

# libraries:
library(tibble)
library(GenomicRanges)
library(rtracklayer)
library(data.table)
library(dplyr)
library(yaml)
library(stringr)
library(report)

## Select outdir
out.dir = paste0("/DATA/projects/DSBrepair/data/R/rs", Date, "/")
dir.create(out.dir, showWarnings = FALSE)


in.dir.date = 20210311
in.dir = paste0("/DATA/projects/DSBrepair/data/R/rs", in.dir.date, "/")

# # out.dir = in.dir
# dir.create(out.dir, showWarnings = FALSE)
```

### Custom functions
Functions used thoughout this script.
```{r}
not_all_na <- function(x) {
  # function to remove NAs
  any(!is.na(x))
}
```

# Data import

## Data loading

The following data exps are either from 1_ParsingQC (mapping: `rs[latestdate]_Analyis_Mapping_RSTP2_2000.txt `), 2_Preprocessing (indels and mean indels ` `) or from the chip means script. All the files are located in `~/mydata/projects/CRISPR_TRIP/20180918_TRIP_5106/indels/data/processed/`

```{r loading}

tib_RSTP2_2000_mutations = readRDS(paste0(in.dir, "total_indel_data.RDS"))
load(paste0(in.dir, "Analyis_Mapping_RSTP2_2000.RData"))
all.mutations.trunc.dt = readRDS(paste0(in.dir, "mutation_frequencies_trunc.RDS"))

# Loading and prepping the ChIP and DamID data.
ChIP_mean_data = readRDS("/DATA/projects/DSBrepair/data/R/cl20200421_ChIP_mean_selection.RDS")
ChIP_meta = readRDS("/DATA/projects/DSBrepair/data/R/cl20200421_ChIP_meta_selection.RDS")
# emsembl.seg.GRCh38 <- import("/DATA/projects/DSBrepair/data/features/homo_sapiens.GRCh38.K562.Regulatory_Build.regulatory_activity.20190329.gff")
# state_cluster <- read.table("/DATA/projects/DSBrepair/data/R/cl20190719_pool_64_10_state.txt", col.names = c("barcode", "cluster"), stringsAsFactors = FALSE)

# Spin states
spin_state <- import("/DATA/projects/DSBrepair/data/K562_SPIN_new_color.bed")

# DNAse peaks
DNAse_Broad =import.bed('/DATA/projects/DSBrepair/data/tracks_hg38/DNase-seq_K562_BroadPeaks_ENCFF828WSI.bed')

# LAD states
LAD.HMM.gr <- import.bed("/home/t.v.schaik/mydata/proj/3D_nucleus/results/ts180110_4DN_DataProcessing/results/HMM/bin-5kb/K562_LMNB1-5kb-combined_HMM.txt.gz")

# ChIP domains
near_dir = '/DATA/projects/DSBrepair/data/tracks_20200327/nearest'

file_list = list.files(near_dir, pattern='20180918.RSTP2', full.names=T)

file.dt = data.table(str_match(file_list, '.*/20180918.RSTP2_(.*).txt'))
names(file.dt) = c('file', 'target')

hist.domains.dt = file.dt[!is.na(file),
                 fread(file, col.names=c('barcode', 'domain', 'distance')),
				 by=c('target')]

clone5bc <- c("AGGGCGTAAAATATTT.B",
                    "TATGGCTGTCGGGTAG.B",
                    "TGTCCCTTAGTACTTT.B",
                    "AGAAAATAATATGACG.B",
                    "CGGCCTGAAGGTCAGG.B",
                    "TTGAACGCGGGCTCGG.B",
                    "GCTAACATCACGAATC.B",
                    "GCGCACCCTTTAATTG.B",
                    "ACTGTCGAGTTGTCCG.B",
                    "CCGGGGACGTATGCAC.B",
                    "TCTTTTGAGGAGCTGA.B",
                    "ATATCGTTGCTGGAGA.B",
                    "CATCCACCACACTTCA.B",
                    "ACCCCTAAAGGCGCTG.B",
                    "ATACTATATTTAACGG.B",
                    "CATTTCTGATCAATAA.B",
                    "GAGCGCGTCACCGGGT.B",
                    "GTACCTCTCGATAGTG.B",
                    "TGGCCAATATTTGTCT.B")
```
That's done, further with the preprossesing of the chromatin data.

```{r}
ChIP_mean_data[pool == "clone_set", ID := paste0(ID, ".B")]
chip_cast.dt = ChIP_mean_data
setnames(chip_cast.dt, "ID", "barcode")
analysis.mapped.integrations.df$end = analysis.mapped.integrations.df$start
```

```{r damid}
rstp2.gr <- makeGRangesFromDataFrame(analysis.mapped.integrations.df, keep.extra.columns = TRUE)

# DamID LAD calling loading and processing
# overlaps
ovl <- findOverlaps(rstp2.gr, LAD.HMM.gr)
mcols(rstp2.gr)$domain_LAD <- NA
mcols(rstp2.gr)$domain_LAD[queryHits(ovl)] <- mcols(LAD.HMM.gr)$name[subjectHits(ovl)]

# DNAse seq
ovl <- findOverlaps(rstp2.gr, DNAse_Broad)
mcols(rstp2.gr)$DNAse_broad <- NA
mcols(rstp2.gr)$DNAse_broad[queryHits(ovl)] <- "DHS"

# Spin state
ovl <- findOverlaps(rstp2.gr, spin_state)
mcols(rstp2.gr)$spin_state <- NA
mcols(rstp2.gr)$spin_state[queryHits(ovl)] <- mcols(spin_state)$name[subjectHits(ovl)]
mcols(rstp2.gr)$spin_color[queryHits(ovl)] <- mcols(spin_state)$itemRgb[subjectHits(ovl)]

# Back to dataframe
rstp2.df <- GenomicRanges::as.data.frame(rstp2.gr, stringsAsFactors = FALSE)
domains.dt <- data.table(rstp2.df[, c("barcode",
                                      "domain_LAD",
                                      "spin_state",
                                      "spin_color",
                                      "DNAse_broad")])

# setnames(domains.dt, "barcode", "ID")
setkey(domains.dt, "barcode")
```

<!-- start Christ -->

```{r domains}
# rep.dt[target=='repliseq', domain:=ifelse(domain=='iDomain', 'Domain', 'iDomain')]
#
hist.domains.dt[target=='repliseq',target:='late_replicating']

chip.domains.dt = hist.domains.dt[!target %in% c('SPIN-10state', 'SPIN-100kb'), ]

chip.domains.dt[distance > 0, domain:='iDomain']

file.spin.dt = data.table(str_match(file_list, '.*/20180918.RSTP2_(SPIN.*).txt'))
names(file.spin.dt) = c('file', 'target')
spin.dt = file.spin.dt[!is.na(file),
                 fread(file, col.names=c('barcode', 'domain', 'distance')),
				 by=c('target')]

# Ruben addition
## temp add clone barcodes
clone_file_list = list.files(near_dir, pattern='clone_set', full.names=T)

clone_file.dt = data.table(str_match(clone_file_list, '.*/clone_set_(.*).txt'))
names(clone_file.dt) = c('file', 'target')

clone_rep.dt = clone_file.dt[!is.na(file),
                             fread(file, col.names=c('barcode', 'domain', 'distance')),
                             by=c('target')]
clone_rep.dt[, barcode := paste0(barcode, ".B")]
clone_rep.dt[, target := gsub("[59]_(.*)", "\\1", target)]

# clone_rep.dt[target=='repliseq', domain:=ifelse(domain=='iDomain', 'Domain', 'iDomain')]

clone_rep.dt[target=='repliseq',target:='late_replicating']


clone_chip.domains.dt = clone_rep.dt[!target %in% c('SPIN-10state', 'SPIN-100kb'), ]

clone_chip.domains.dt[distance > 0, domain:='iDomain']

chip.domains.dt = rbind(clone_chip.domains.dt,
                        chip.domains.dt, spin.dt) %>%
                        distinct(barcode, target, .keep_all = TRUE)

chip.domains.dt = mutate(chip.domains.dt, target = paste0(target, "_domain")) %>% 
  dplyr::select(-distance) %>%
  data.table()
chip.domains.dt[domain == "iAD", domain := 'iDomain']
chip.domains.dt[domain == "AD", domain := 'Domain']
chip.domains.cast.dt = tidyr::spread(chip.domains.dt, target, domain)
```

<!-- end Christ -->



```{r data merge}
chip_domains.dt = chip_cast.dt[chip.domains.cast.dt, on = "barcode"]
chip_domains.dt = chip_domains.dt[domains.dt, on = "barcode"]

setnames(chip_domains.dt, c("domain_LAD", "DNAse_broad", "SPIN-100kb_domain", "SPIN-10state_domain"),
         c("LAD", "DHS_domain", "SPIN-100kb", "SPIN-10state"))

chip_domains.dt[, pool := NULL]

# Merge on barcode (cell line is not required - however not that the chromatin data is not accurate for the KO cell lines)
trip_tib = left_join(tib_RSTP2_2000_mutations, chip_domains.dt, by = c("barcode")) %>% 
  as_tibble() %>% 
  distinct(exp_pool, mutation, barcode, .keep_all = TRUE) # remove any duplications that might have occured with the merge, while keeping all the columns.
trip_tib <- trip_tib[!is.na(trip_tib$exp), ] # remove NAs if applicable
```

### Remove K9me3 data
```{r removing K9me3 dataa}
trip_tib <- dplyr::select(trip_tib, !c("H3K9me3"))
```


# Conclusions
```{r}
# trip_tib_1000 <- trip_tib %>% filter(binsize == 1000) %>% select_if(not_all_na)
trip_tib_2000 <- trip_tib %>% filter(binsize == 2000) %>% select_if(not_all_na)

# Same for the mutations plotting file
trip_tib_mut = left_join(all.mutations.trunc.dt, chip_domains.dt, by = c("barcode")) %>% as_tibble()
trip_tib_mut <- trip_tib_mut[!is.na(trip_tib_mut$exp), ]
# trip_tib_mut_1000 <- trip_tib_mut %>% filter(binsize == 1000) %>% select_if(not_all_na)
trip_tib_mut_2000 <- trip_tib_mut %>% filter(binsize == 2000) %>% select_if(not_all_na)

## Export this data table
# Ratios
# save(trip_tib_1000, file = paste0(out.dir, "RSTP2_IndelRatios_Chromatin_1kb.RData"))
saveRDS(trip_tib_2000, file = paste0(out.dir, "RSTP2_IndelRatios_Chromatin_2kb.RDS"))

# Mutations truncated
# save(trip_tib_mut_1000, file = paste0(out.dir, "RSTP2_Indel_Chromatin_1kb.RData"))
saveRDS(trip_tib_mut_2000, file = paste0(out.dir, "RSTP2_Indel_Chromatin_2kb.RDS"))
```

# Bibliography
```{r citations}
cite_packages()
```


# Session Info
```{r}
paste("Run time: ",format(Sys.time()-StartTime))
getwd()
date()
sessionInfo()
# rm(list = ls(all.names = TRUE)) #will clear all objects includes hidden objects.
gc() #free up memory and report the memory usage.
```
